_posts/2016-08-02-what-does-RESTful-really-mean.md

```diff
@@ -0,0 +1,147 @@
+
+One of the drawbacks of having a stateless server is decreased network performance as all the needed data has to be sent in each request
+
+* Cacheable
+
+REST applications are web systems; therefore, clients and intermediaries can cache responses. Responses themselves must be defined as cacheable, or not, to prevent clients from reusing stale data that could reduce *reliability* if stale data in the cache differs significantly from the data that would have been generated the request been handled by the server. Caching could eliminate some client-server interaction, thus improving *scalability*, *efficiency* and user-perceived *performance* by reducing average latency.
+
+* Uniform interface
+
+Using a uniform interface simplifies and decouples the architecture and favours the independent evolution of different parts. As explained later on in this post, URIs, resources and hypermedia help to produce a standard interface that improves *visibility* of interactions, *simplifies* the overall system architecture and induces independent evolution. The trade-off is that it degrades *efficiency* since information is transferred in a standard format rather one which is particular to an application's needs.
+
+* Layered system
+
+Using a layered system reduces complexity by constraining component behaviour such that each element cannot access beyond its immediate layer. Favours substrate independence by restricting knowledge of other parts of the system. Layers can encapsulate legacy components and protect new services from legacy clients. Intermediaries can be used to improved *scalability* by enabling load balancing across networks. The main trade-off is that layered system add overhead and latency to the data processing, therefore, reducing user-perceived *performance*.
+
+* Code-On-Demand (Optional)
+
+REST allows clients to extend their functionality by downloading and executing code scripts. This *simplifies* clients and improves *extensibility*. On the other hand, it reduces *visibility*, that's why is only an optional constraint.
+
+## Elements
+
+REST have several elements in its toolbox to build stateless, scalable and simple web APIs.
+
+* HTTP
+* Resources
+* URIs
+* Hypermedia
+
+### HTTP - Document transfer application protocol
+
+REST is usually used along with HTTP as its transfer protocol as it offers several advantages. Among them are HTTP verbs, status codes and headers.
+
+#### Verbs
+
+Instead of defining new verbs for every possible behaviour in our web service. HTTP introduces a standard set of verbs to handle similar situations in the same way, therefore, removing unnecessary variation and creating a more intuitive API. Each verb has a different combination of two properties that make them suitable for different scenarios.
```

Comment:
I'd say that therefor is redundant in this sentence, less words is usually better. But this is a matter of style, up to you :)